src/aggregate/aggregate_exec.c

```diff
@@ -1137,82 +1139,88 @@ typedef struct {
 
 static void cursorRead_ctx(CursorReadCtx *cr_ctx) {
   RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(cr_ctx->bc);
-  RedisModule_Reply _reply = RedisModule_NewReply(ctx), *reply = &_reply;
-  cursorRead(reply, cr_ctx->cid, cr_ctx->count, true);
-  RedisModule_EndReply(reply);
+  cursorRead(ctx, cr_ctx->cid, cr_ctx->count, true);
   RedisModule_FreeThreadSafeContext(ctx);
   RedisModule_BlockedClientMeasureTimeEnd(cr_ctx->bc);
   RedisModule_UnblockClient(cr_ctx->bc, NULL);
   rm_free(cr_ctx);
 }
 
 /**
- * FT.CURSOR READ {index} {CID} {COUNT} [MAXIDLE]
- * FT.CURSOR DEL {index} {CID}
- * FT.CURSOR GC {index}
+ * FT.CURSOR READ {index} {CID} [COUNT {COUNT}]
  */
-int RSCursorCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  if (argc < 4) {
+int RSCursorReadCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc < 4 || argc > 6) {
     return RedisModule_WrongArity(ctx);
   }
 
-  const char *cmd = RedisModule_StringPtrLen(argv[1], NULL);
-  long long cid = 0;
-  // argv[0] - FT.CURSOR
-  // argv[1] - subcommand
-  // argv[2] - index
-  // argv[3] - cursor ID
-
+  long long cid;
   if (RedisModule_StringToLongLong(argv[3], &cid) != REDISMODULE_OK) {
-    RedisModule_ReplyWithError(ctx, "Bad cursor ID");
-    return REDISMODULE_OK;
+    return RedisModule_ReplyWithError(ctx, "Bad cursor ID");
   }
 
-  RedisModule_Reply _reply = RedisModule_NewReply(ctx), *reply = &_reply;
-
-  if (strcasecmp(cmd, "READ") == 0) {
-    long long count = 0;
-    if (argc > 5) {
-      // e.g. 'COUNT <timeout>'
-      // Verify that the 4'th argument is `COUNT`.
-      const char *count_str = RedisModule_StringPtrLen(argv[4], NULL);
-      if (strcasecmp(count_str, "count") != 0) {
-        RedisModule_ReplyWithErrorFormat(ctx, "Unknown argument `%s`", count_str);
-        RedisModule_EndReply(reply);
-        return REDISMODULE_OK;
-      }
-
-      if (RedisModule_StringToLongLong(argv[5], &count) != REDISMODULE_OK) {
-        RedisModule_ReplyWithErrorFormat(ctx, "Bad value for COUNT: `%s`", RedisModule_StringPtrLen(argv[5], NULL));
-        RedisModule_EndReply(reply);
-        return REDISMODULE_OK;
-      }
+  long long count = 0;
+  if (argc > 4) {
+    // e.g. 'COUNT <chunk size>'
+    // Verify that the 4'th argument is `COUNT`.
+    if (strcasecmp(RedisModule_StringPtrLen(argv[4], NULL), "count") != 0) {
+      return RedisModule_ReplyWithError(ctx, "Unknown argument at offset 4");
```

Comment:
maybe the number 4 should be a constant and then be printed in the error message?